### PR TITLE
Iv 1342 rightscale mirrors

### PR DESCRIPTION
**Pull request by psschroeter**
See file: simulates RightScale frozen mirrors on dashboard. By default this functionality is turned off.

### CODE REVIEW

File: rll/attachments/RPM-GPG-KEY-EPEL-7 [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+9Ism3CYydpy0GLRA5wo4fPB5uLdUhLEUDvh2KK//fMjja3o0L+SNz8N0aDZyn5Ax
+CU9RB3EHcTecFgoy5umRj99BZrebR1NO+4gBrivIfdvD4fJNfNBHXwhSH9ACGCNv
+HnXVjHQF9iHWApKkRIeh8Fr2n5dtfJEF7SEX8GbX7FbsWo29kXMrVgNqHNyDnfAB
+VoPubgQdtJZJkVZAkaHrMu8AytwT62Q4eNqmJI1aWbZQNI5jWYqc6RKuCK6/F99q
+thFT9gJO17+yRuL6Uv2/vgzVR1RGdwVLKwlUjGPAjYflpCQwWMAASxiv9uPyYPHc
+ErSrbRG0wjIfAR3vus1OSOx3xZHZpXFfmQTsDP7zVROLzV98R3JwFAxJ4/xqeON4
+vCPFU6OsT3lWQ8w7il5ohY95wmujfr6lk89kEzJdOTzcn7DBbUru33CQMGKZ3Evt
+RjsC7FDbL017qxS+ZVA/HGkyfiu4cpgV8VUnbql5eAZ+1Ll6Dw==
+=hdPa
+-----END PGP PUBLIC KEY BLOCK-----
```
Comment (lopaka) [inline rll/attachments/RPM-GPG-KEY-EPEL-7]: Perhaps add a newline at end of file?

File: rll/rightscale-mirrors.sh [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+fi
+
+RUBYGEMS_FREEZE_DATE=${RUBYGEMS_FREEZE_DATE//-/}
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  echo "Since RUBYGEMS_FREEZE_DATE is not set, not configuring Rubygems mirror"
+elif [[ ! "$RUBYGEMS_FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ ! "$RUBYGEMS_FREEZE_DATE" = "latest" ]]; then
+  echo "If RUBYGEMS_FREEZE_DATE is set, it must be either 'latest' or in the format YYYY-MM-DD"
+  exit 1
+fi
+today=$(date +"%Y%m%d")
```
Comment (lopaka) [inline rll/rightscale-mirrors.sh]: `today` is already set on line 38.  Can we set it in a single place, perhaps towards the top.

File: rll/rightscale-mirrors.sh [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+if [[ "$RUBYGEMS_FREEZE_DATE" =~ ^[0-9]{8}$ ]] && [[ $RUBYGEMS_FREEZE_DATE -gt $today ]]; then
+  echo "RUBYGEMS_FREEZE_DATE can't be from a date in the future"
+  exit 1
+fi
+
+if [[ -n "$FREEZE_DATE" ]] || [[ -n "$RUBYGEMS_FREEZE_DATE" ]]; then
+  if [[ -z "$MIRROR_HOST" ]]; then
+    echo "MIRROR_HOST cannot be blank, possible values are cf-mirror.rightscale.com or ENV:RS_ISLAND"
```
Comment (lopaka) [inline rll/rightscale-mirrors.sh]: Shouldn't this condition `exit 1` as well?

File: rll/rightscale-mirrors.sh [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+fi
+rubygems_freeze_date_msg=$RUBYGEMS_FREEZE_DATE
+if [[ -z "$RUBYGEMS_FREEZE_DATE" ]]; then
+  rubygems_freeze_date_msg='(unconfigured)'
+fi
+echo "Distro: $distro"
+echo "Distro version: $distro_ver"
+echo "Architecture: $arch"
+echo "OS Repositories Freeze Date: $freeze_date_msg"
+echo "Rubygems Freeze Date: $RUBYGEMS_FREEZE_DATE"
```
Comment (lopaka) [inline rll/rightscale-mirrors.sh:110]: Should we fail if any of these variables are not set correctly?  I don't see any failure catches in the conditions above.

Comment (psschroeter) [inline rll/rightscale-mirrors.sh:110]: So we do fail at some point below (line 189) -- if the value of distro isn't centos/redhat/ubuntu then we error out in a case statement.

Comment (lopaka) [inline rll/rightscale-mirrors.sh:110]: :+1:

**Comment by tve** [comment]
I'm not OK with this PR. I mentioned that in the ticket, sorry for not commenting here. I believe it's time to start a 'compat' or similarly named "cookbook" and put legacy compatibility stuff there. This frozen mirror stuff should not be seen as part of RL10 and we need to be able to deprecate them at some point, so I'd like this script and others to be clearly separated from the core RL10 stuff.
Having a compat cookbook also allows services and customer to contribute other things. For example, could have a script that defines some global env vars that RL10 doesn't have.

**Pull request merged**